### PR TITLE
[WIP] add blog post leveraging git and github for custom tools

### PR DESCRIPTION
**Pull request by gregbunce**
whoa!

### CODE REVIEW

File: _posts/2018-05-15-leverage-git-to-manage-lifecyle-of-arcgis-custom-tools.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+### Know the APIs
+The ArcObjects Desktop APIs are relatively stable and the Interfaces do not change.  When additional functionality is warranted, ESRI creates a new Interface with the same name but with an appended, incrementing number.  For example, over the years the IEditor Interface has expanded to add additional functionality, yet IEditor remains unchanged.  To take advantage of this new functionality you would use IEditor2 or IEditor3 (if it's available in the SDK you're working with).  The benefit of this model is that IEditor is same weather you're using the Desktop 9.2 SDK or 10.6 SDK.  In terms of managing universal source code, this is great. The downside is that it makes the ArcObjects Desktop API a bit verbose and messy to navigate with all the variations of Interfaces.  But, if you know the APIs for the versions you're working with, your code is extremely reusable.
```
Comment (steveoh) [inline _posts/2018-05-15-leverage-git-to-manage-lifecyle-of-arcgis-custom-tools.md]: `weather` -> `whether`

File: _posts/2018-05-15-leverage-git-to-manage-lifecyle-of-arcgis-custom-tools.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+Things are a bit different on the ArcGIS Pro platform.  With the Pro SDK, the API can change with the release of a major version (from Pro 1.x to Pro 2.x), potentially resulting in breaking changes to your code.  Though, if you're developing tools using the standard Extensions and Interfaces (Editing, Geoprocessing, etc), I don't see this being an issue for code reusability.  Again, knowing the APIs for the versions you are working with is key to allowing you to maintain universal source code.  A good starting is to always read the release notes for each [major version](http://pro.arcgis.com/en/pro-app/get-started/release-notes.htm).
```
Comment (steveoh) [inline _posts/2018-05-15-leverage-git-to-manage-lifecyle-of-arcgis-custom-tools.md]: `A good starting` sounds funny. Are you missing some words?

File: _posts/2018-05-15-leverage-git-to-manage-lifecyle-of-arcgis-custom-tools.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+The idea is to use Git and GitHub to manage your code in a way that you maintain a single code base for all versions of your tool.  You'll still need to build separate development machines that correspond to the SDK versions you are supporting.  I recommend building, and then making all code modifications, on the development machine with the lowest version you are supporting.  In my case, that's ArcGIS Desktop 10.4 - so that's where I do all my development and code modifications.  When this code is stable, you 'push' the changes/modifications to the remote repository (GitHub).  Essentially, in this scenario, the remote repository contains the stable version of the code and then also acts as the messenger to the other development machines.  You can then 'pull' the changes from the remote repository to your development machine that is supporting the higher version (in my case, ArcGIS Desktop 10.5).  We'll talk more about 'push' and 'pull' in the 'Common Git Commands' section below.
```
Comment (steveoh) [inline _posts/2018-05-15-leverage-git-to-manage-lifecyle-of-arcgis-custom-tools.md]: turning `Common Git Commands` into an anchor link would be nice.

File: _posts/2018-05-15-leverage-git-to-manage-lifecyle-of-arcgis-custom-tools.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+In my dev environment, this workflow works because I'm building the tools on the lowest version of the SDK that I'm supporting, so I can rely on the API libraries being present in the higher versions of the SDK (see 'Know the APIs' for more info).  As far as getting around the version-specific SDK dependencies (see 'Dealing with SDK Dependencies' for more info) you will want to set the 'specific version' to 'false' for all your ESRI Reference Libraries (in Visual Studio).  This will tell the .NET compiler, when building the solution, that the reference libraries are relative to development machine (and not hard coded to specific version-based directory).
```
Comment (steveoh) [inline _posts/2018-05-15-leverage-git-to-manage-lifecyle-of-arcgis-custom-tools.md]: again, anchor link `Know the APIs`, `Dealing with SDK Dependencies` etc

File: _posts/2018-05-15-leverage-git-to-manage-lifecyle-of-arcgis-custom-tools.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+A final set-up note for this workflow is that you will want the local repository (managed by Git) on all of your development machines to be pointing to the same remote repository (your GitHub repository for this project).  This is how you move your local changes around and how you keep all development environments in sync with the 'master' (or stable version).  The easiest way to set this up is to 'clone' the GitHub repository onto the development machines.  For example, If I want to begin support my existing ArcGIS tools on Desktop 10.6, I'll build a 10.6-specific development machine and then 'clone' the corresponding GitHub repository that I'm maintaining on the 10.4 machine.  This brings down the remote repository code (in its current state) to my local machine.  If I'm still supporting 10.4 then I'll continue to make all code modifications there and then 'pull' the changes to the 10.5 and 10.6 machines as needed.  This is all done via common Git commands (see the 'Git Cheatsheet' below).
```
Comment (steveoh) [inline _posts/2018-05-15-leverage-git-to-manage-lifecyle-of-arcgis-custom-tools.md]: `If I want to begin support my` sounds funny. 
anchor link `git cheatsheet`

**Comment by steveoh** [comment]
just let me know when everyone is happy and I'll push the buttons.

**Comment by atribolet** [comment]
haha thanks for the collaboration! I'll take another look at the article with the changes, and I'll keep you posted once I'm finished with the second review :) 

Thanks!

**Comment by steveoh** [comment]
15 days! It's time...

**Pull request merged**